UserApi/UserApi/Caching/GenericDistributedCache.cs

```diff
@@ -47,5 +46,19 @@ public async Task<T> GetOrAddAsync<T>(string key, Func<Task<T>> factory)
 
             return result;
         }
+
+        public async Task RefreshCacheAsync<T>(Func<Task<T>> factory)
+        {
+            var key = factory.ToString();
+
+            var result = await factory();
+
+            var bytes = Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(result));
+
+            await _distributedCache.SetAsync(key, bytes, new DistributedCacheEntryOptions
+            {
+                AbsoluteExpirationRelativeToNow = TimeSpan.FromHours(3)
+            });
```

Comment:
This can be refactored out so it can be called from here and GetOrAddAsync

Reply:
Cool let me add that

Reply:
Sorted